crates/distribution-types/src/lib.rs

```diff
@@ -119,6 +119,7 @@ impl std::fmt::Display for InstalledVersion<'_> {
 /// The location can be index, url or path (wheel) or index, url, path or git (source distribution)
 #[derive(Debug, Clone)]
 pub enum Dist {
+    Installed(InstalledDist),
```

Comment:
I think adding this to `Dist` is not quite right... I feel like we need some other enum on top of this, like `ResolvedDist` that can either be a remote distribution (`Dist`) or an installed distribution.

Reply:
Yeah I think I share your feeling. Bigger change but might be a better level...